common/cards/card-plugins/effects/turtle-shell.js

```diff
@@ -39,6 +37,76 @@ class TurtleShellEffectCard extends EffectCard {
 			description:
 				'Attach to any of your afk hermits. When the hermit is made active, it prevents any damage for its first turn and then is discarded.',
 		})
+
+		this.instances = {}
+	}
+
+	/**
+	 * @param {GameModel} game
+	 * @param {import('../../../types/cards').CardPos} pos
+	 */
+	canAttach(game, pos) {
+		const {currentPlayer} = game.ds
+
+		if (pos.slot.type !== 'effect') return 'INVALID'
+		if (pos.playerId !== currentPlayer.id) return 'INVALID'
+
+		if (!pos.row?.hermitCard) return 'NO'
+
+		// turtle shell addition - hermit must be inactive to attach
+		if (!(currentPlayer.board.activeRow !== pos.rowIndex)) return 'NO'
+
+		return 'YES'
+	}
+
+	/**
+	 * @param {GameModel} game
+	 * @param {string} instance
+	 * @param {import('../../../types/cards').CardPos} pos
+	 */
+	onAttach(game, instance, pos) {
+		const instanceKey = this.getKey(instance)
+		pos.player.custom[instanceKey] = false
+
+		pos.otherPlayer.hooks.onAttack[instance] = (attack) => {
+			if (
+				pos.player.custom[instanceKey] === true &&
+				attack.target.row.effectCard?.cardInstance === instance
+			) {
+				attack.multiplyDamage(0).lockDamage()
+			}
+			return attack
+		}
+
+		pos.otherPlayer.hooks.afterAttack[instance] = () => {
+			if (pos.player.custom[instanceKey] === true) {
+				discardCard(game, {cardId: this.id, cardInstance: instance})
+			}
+		}
+
+		pos.otherPlayer.hooks.turnStart[instance] = () => {
```

Comment:
I see a slight problem with this that means we may need another hook - what if at the start of your turn you have no active hermit because you just lost a life, and then you switch to our hermit?